### PR TITLE
chore(ts-migration): updates analytics widgets and stories

### PR DESCRIPTION
**Pull request by nunomaduro**
This pull request adjusts widgets and stories in analytics to be similar to the other already migrated widgets and stories;

### CODE REVIEW

File: src/widgets/analytics/analytics.ts [hunk 1]
```diff
@@ -1,42 +1,77 @@
...
-  triggerOnUIInteraction: boolean;
+  triggerOnUIInteraction?: boolean;
+
   /**
-   * @default false
+   * Triggers `pushFunction` when InstantSearch is initialized. This means
+   * the `pushFunction` might be called even though the user didn’t perfom
+   * any search-related action.
+   *
+   * @default true
```
Comment (nunomaduro) [inline src/widgets/analytics/analytics.ts:50]: This was `default` false before, but was wrong.

File: src/widgets/analytics/analytics.ts [hunk 2]
```diff
@@ -1,42 +1,77 @@
...
+  pushPagination?: boolean;
 };
 
 const withUsage = createDocumentationMessageGenerator({ name: 'analytics' });
 
-function analytics(
-  {
+export type AnalyticsWidget = WidgetFactory<{}, AnalyticsWidgetParams>;
+
+const analytics: AnalyticsWidget = function analytics(widgetOptions) {
```
Comment (eunjae-lee) [inline src/widgets/analytics/analytics.ts:66]:
```suggestion
const analytics: AnalyticsWidget = function analytics(widgetParams) {
```

File: src/widgets/analytics/analytics.ts [hunk 3]
```diff
@@ -1,42 +1,77 @@
...
+  const {
     pushFunction,
     delay = 3000,
     triggerOnUIInteraction = false,
     pushInitialSearch = true,
     pushPagination = false,
-  }: AnalyticsWidgetParams = {} as AnalyticsWidgetParams
-): Widget {
+  } = widgetOptions || ({} as typeof widgetOptions);
```
Comment (eunjae-lee) [inline src/widgets/analytics/analytics.ts:73]:
```suggestion
  } = widgetParams || ({} as typeof widgetParams);
```

Comment (nunomaduro) [reply src/widgets/analytics/analytics.ts]: Fixed.

File: src/widgets/analytics/analytics.ts (hunk 2 above)
Comment (nunomaduro) [reply src/widgets/analytics/analytics.ts]: Fixed.

**Pull request merged**